optuna/warning_decorators.py

```diff
@@ -0,0 +1,102 @@
+import functools
+from typing import Any
+from typing import Callable
+import warnings
+
+
+_DOCSTRING_TEMPLATE = """
```

Comment:
The below might be better than this considering that it's possible that we implement decorators such as `deprecated`.
```suggestion
_EXPERIMENTAL_DOCSTRING_TEMPLATE = """
```